### ISSUE TITLE
Using b.require with react and react-dom results in an error

### ISSUE THREAD

**Issue by casr** [browserify/common-shakeify#21]
> browserify -r react -r react-dom -p tinyify

    /.../node_modules/tinyify/node_modules/common-shakeify/index.js:222
      return row[kDuplicates] || []
                ^
    
    TypeError: Cannot read property 'Symbol(duplicates)' of undefined
        at getDuplicates (/.../node_modules/tinyify/node_modules/common-shakeify/index.js:222:13)
        at analyzer.modules.forEach (/.../node_modules/tinyify/node_modules/common-shakeify/index.js:121:21)
        at Map.forEach (<anonymous>)
        at DestroyableTransform.onend [as _flush] (/.../node_modules/tinyify/node_modules/common-shakeify/index.js:118:22)
        at DestroyableTransform.prefinish (/.../node_modules/readable-stream/lib/_stream_transform.js:138:10)
        at emitNone (events.js:106:13)
        at DestroyableTransform.emit (events.js:208:7)
        at prefinish (/.../node_modules/readable-stream/lib/_stream_writable.js:619:14)
        at finishMaybe (/.../node_modules/readable-stream/lib/_stream_writable.js:627:5)
        at endWritable (/.../node_modules/readable-stream/lib/_stream_writable.js:638:3)

It appears to only happen when I use them both; if I `-r` them separately then it works fine. I tried it with some simpler packages such as `once` and `xtend` together and that appeared to work just fine.

The package.json:

    {
      "private": true,
      "scripts": {
        "test": "browserify -r react -r react-dom -p tinyify"
      },
      "dependencies": {
        "react": "16.6.3",
        "react-dom": "16.6.3"
      },
      "devDependencies": {
        "browserify": "16.2.3",
        "tinyify": "2.4.3"
      }
    }

**Comment by casr** [comment]
Might be related with the other work I was doing over at https://github.com/browserify/browserify/pull/1874 which for now I fixed it with more of a workaround.

From what I can gather, I think the underlying issue is that `row.id/row.file` is pushed through to `module-deps` to be resolved but `row.file` does not get updated. I tried a version where in `b.require` I had a synchronous call to `bresolve` and set the `row.file` properly prior to pushing it on to the pipeline. It sorta worked but messed up some of the expose tests (I think).
